WeatherApp_Obj-C/WeatherApp_Obj-C/Screens/HomeScreen/HomeScreenViewController.m

```diff
@@ -20,17 +32,109 @@ - (void)viewDidLoad {
+    if (self) {
+        sunny = 0;
+        cloudy = 0;
+        raining = 0;
+        items = [[NSMutableArray alloc] init];
+    }
+    return self;
+}
+
+-(void)fetchData {
+    items = [[WeatherRepository new] fetchCDData];
+    [self.tableView reloadData];
 }
 
 - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:animated];
     [self.navigationController setNavigationBarHidden:YES animated:YES];
+    [self fetchData];
+    [self setupWeatherCounter];
 }
+
+-(void)setupTableView {
+    self.tableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
+    [self.tableView registerNib:[UINib nibWithNibName:NSStringFromClass([HomeScreenCell class]) bundle:nil] forCellReuseIdentifier:NSStringFromClass([HomeScreenCell class])];
+    self.tableView.rowHeight = UITableViewAutomaticDimension;
+}
+
+-(void)setupWeatherCounter {
+    [self resetWeatherCounter];
+    for (int i = 0; i < items.count; i++) {
+        NSString *main = items[i].mainEvent;
+        if ([main containsString:@"rain"] || [main containsString:@"thunderstorm"] || [main containsString:@"drizzle"]) {
+            raining++;
+        } else if ([main containsString:@"clouds"]) {
+            cloudy++;
+        } else if ([main containsString:@"clear"]) {
+            sunny++;
+        }
+        self.sunnyCounter.text = [NSString stringWithFormat:@"%d", sunny];
+        self.rainingCounter.text = [NSString stringWithFormat:@"%d", raining];
+        self.cloudyCounter.text = [NSString stringWithFormat:@"%d", cloudy];
+    }
+}
+
+-(void)resetWeatherCounter {
+    sunny = 0;
+    cloudy = 0;
+    raining = 0;
+}
+
+#pragma TableView delegate methods
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return items.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    HomeScreenCell *cell = [self.tableView dequeueReusableCellWithIdentifier:NSStringFromClass([HomeScreenCell class])];
+    cell.cellName.text = items[indexPath.row].name;
+    cell.regionName.text = items[indexPath.row].region;
+    cell.temperature.text = [NSString stringWithFormat:@"%.01f°C", items[indexPath.row].temperature];
+    cell.rain.text = [NSString stringWithFormat:@"%.01f%%", items[indexPath.row].rain];
+    cell.wind.text = [NSString stringWithFormat:@"%.01f%%", items[indexPath.row].wind];
+    cell.timeSinceAdded.text = [self calculateTime:items[indexPath.row].time];
+    NSData * imageData = [[NSData alloc] initWithContentsOfURL: [NSURL URLWithString: items[indexPath.row].imgUrl]];
+    cell.weatherImage.image = [UIImage imageWithData: imageData];
+    cell.colorBackground.backgroundColor = [self loadBackgroundColor:items[indexPath.row].mainEvent];
+    return cell;
+}
+
+-(UIColor*)loadBackgroundColor:(NSString*)main {
+    if ([main containsString:@"rain"] || [main containsString:@"thunderstorm"] || [main containsString:@"drizzle"]) {
+        raining++;
```

Comment:
Should these values be incremented here? They will get incremented every time the user scrolls through the table view

Reply:
This was solved in the later PR